fastly-edge-assignments/src/handlers/handler_assignments.rs

```diff
@@ -129,7 +129,7 @@ pub fn handle_assignments(mut req: Request) -> Result<Response, Error> {
 
     // Create an HTTP OK response with the assignments
     Response::from_status(StatusCode::OK)
-        .with_body_json(&precomputed_configuration)
+        .with_body_json(&precomputed_configuration.obfuscate())
```

Comment:
kinda proud that edge service change is 1 word 😎 

cc @leoromanovsky

Reply:
won't this break existing clients (which don't support obfuscated?)

Reply:
That's the best kind of change 😄 

FYI the JS browser client does not currently support obfuscation decoding and is deployed in production in our FE product on `eppo.cloud`. 

I am thinking whether we should do a conditional obfuscation, but that raises the question of why we would ever want "not obfuscated". Since we still don't have any public customers for this I would argue we could:

* Keep the code as-is
* Modify the JS-client to only accept obfuscated configuration; reject currently published version as not workable.

Reply:
I would say that it's too early to treat precomputed configs as "in production." There's no real users, rolled out implementation doesn't follow the spec, and it's already broken by fixes in https://github.com/Eppo-exp/eppo-multiplatform/pull/110. I was a bit surprised to learn that JS SDK support was already released given that implementation is so off the spec

Reply:
> but that raises the question of why we would ever want "not obfuscated".

"Evaluating" non-obfuscated configuration is significantly faster. That might be one reason

Reply:
Yes we SHOULD modify the js-client to only accept obfuscated precomputed flags. I agree with that direction!

Some options for moving forward:
1. Stop dogfooding in the Eppo app (switch back to the normal config-evaluation-based client) so we can make this change and deprecate the current js-client-sdk version
2. Version the edge compute URL so that obfuscated and unobfuscated precomputed flags have different endpoints, and release a new js-client-sdk version that points to the new endpoint

Reply:
> "Evaluating" non-obfuscated configuration is significantly faster. That might be one reason

What would be the bottleneck? Base64 decoding?

Reply:
@sameerank there's a couple of changes in #110 that break unobfuscated format as well, so there will still be some breakage in #2:
- logging fields (`allocation_key`, `variation_key`, `extra_logging`) are now optional. Previously, the service was skipping allocations with `doLog: false`
- JSON values are strings (same as UFC). Previously they were shipped as-is

Reply:
Let's go with option 1 then. We'll pause dogfooding and start again when the js-client-sdk is updated to use obfuscated precomputed assignments

Reply:
I agree we don't need to support unobfuscated just for our dogfooding. Pausing and restarting makes sense.

One use case for non-obfuscated responses is for live subject evaluation. So I think we still want the edge function to be able to do it, such as when we hit it with a JWT rather than SDK key

However, no _SDK_ needs to be able to request and process a non-obfuscated response I don't think.

Reply:
Ok we can create a separate handler for the JWT auth and it can serve not-obfuscated if needed. Lets do this after we move it to the private repo please.